contracts/Locker.sol

```diff
@@ -0,0 +1,201 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./library/BoringMath.sol";
+import "./interface/ILocker.sol";
+
+contract Locker is ILocker, ReentrancyGuard, Ownable {
+    using BoringMath for uint256;
+    using SafeERC20 for IERC20;
+
+    struct Balances {
+        uint112 locked;
+        uint112 boosted;
+        uint32 nextUnlockIndex;
+    }
+
+    struct LockedBalance {
+        uint112 amount;
+        uint32 unlockTime;
+    }
+
+    struct Epoch {
+        uint224 supply; //epoch boosted supply
+        uint32 date; //epoch start date
+    }
+
+    //token constants
+    IERC20 public constant stakingToken = IERC20(0x4e3FBD56CD56c3e72c1403e103b45Db9da5B9D2B); //cvx
+
+    bool public isShutdown = false;
+
+    // Duration that rewards are streamed over
+    uint256 public constant REWARDS_DURATION = 7 days;
+
+    // Duration of lock/earned penalty period
+    uint256 public constant lockDuration = REWARDS_DURATION * 16;
+
+    uint256 public constant denominator = 10000;
+
+    //supplies and epochs
+    uint256 public lockedSupply;
+    Epoch[] public epochs;
+
+    //staking
+    uint256 public minimumStake = 10000;
+    uint256 public maximumStake = 10000;
+    address public stakingProxy;
+    address public constant cvxcrvStaking = address(0x3Fe65692bfCD0e6CF84cB1E7d24108E434A7587e);
+    uint256 public constant stakeOffsetOnLock = 500; //allow broader range for staking when depositing
+
+    //mappings for balance data
+    mapping(address => Balances) public balances;
+    mapping(address => LockedBalance[]) public userLocks;
+
+    constructor() {
+        uint256 currentEpoch = (block.timestamp / REWARDS_DURATION) * REWARDS_DURATION;
```

Comment:
Can set the struct field/member directly w/o this variable since `currentEpoch` isn't being used anywhere else

Reply:
Fixed